internal/gengapic/genrest.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2021  Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gengapic
+
+import (
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/googleapis/gapic-generator-go/internal/errors"
+	"github.com/googleapis/gapic-generator-go/internal/pbinfo"
+)
+
+func (g *generator) restClientInit(serv *descriptor.ServiceDescriptorProto, servName string, imp pbinfo.ImportSpec, hasRPCForLRO bool) {
+	p := g.printf
+
+	lowcaseServName := lowerFirst(servName + "RESTClient")
```

Comment:
Kind of an annoying nit, sorry: having `rESTClient` would be unfortunate. Would it be possible to say `if serName == "" { servName = "restClient" }` or something of the like?

We should probably do the same with `gRPCClient` -> `grpcClient` , although funnily enough the lower first form is the proper casing for the actual product name... @tbpg WDYT?

Reply:
`restClient` would be nice if `servName == ""`.

I'm good with `gRPC`.

Reply:
Neither client is exported, so I think we don't absolutely need to be as friendly as possible with the name.
I am in general against having a special case for `servName == ""`, but the generated output does look better.
Going to leave `gRPC` alone, since it is vaguely idiomatic.

Reply:
SGTM